crates/sui-json-rpc/src/name_service.rs

```diff
@@ -90,6 +98,25 @@ impl Domain {
     pub fn depth(&self) -> u8 {
         self.labels.len() as u8
     }
+
+    /// Formats a domain into a string based on the available output formats.
+    /// The default separator is `.`
+    pub fn format(&self, format: DomainFormat) -> String {
+        let mut labels = self.labels.clone();
+        let sep = &ACCEPTED_SEPARATORS[0].to_string();
+        labels.reverse();
+
+        if format == DomainFormat::Dot {
+            return labels.join(sep);
+        };
+
+        // SAFETY: This is a safe operation because we only allow a
+        // domain's label vector size to be >= 2 (see `Domain::from_str`)
+        let _tld = labels.pop();
```

Comment:
Is it worth testing that this thing is actually "sui" or does it not matter?

Reply:
Shouldn't matter here! That's something being checked on the endpoints querying this. 
(By design, you could have domains with different tlds).